### PR TITLE
[WIP] Add FlatCache lib and use for Notifications

### PR DESCRIPTION
**Pull request by rnystrom**
WIP for #597

Putting this in its own local lib because the build times for running a single test are awful. Breaking this stuff out and working in test libs is a wonderful boost to productivity.

### CODE REVIEW

File: Local Pods/FlatCache/FlatCache.podspec [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Pod::Spec.new do |spec|
+  spec.name         = 'FlatCache'
+  spec.version      = '0.1.0'
+  spec.license      = { :type => 'MIT' }
+  spec.homepage     = 'https://github.com/tonymillion/Reachability'
```
Comment (Sherlouk) [inline Local Pods/FlatCache/FlatCache.podspec]: Ooh I love our new homepage 😉

Comment (rnystrom) [reply Local Pods/FlatCache/FlatCache.podspec]: lol whoops. Copy paste ftw

**Comment by rnystrom** [comment]
simulator-screen-shot---iphone-8---2017-10-21-at-20 13 15

**Comment by rnystrom** [comment]
Going to merge this so I can send out a build and get some feedback

**Comment by BasThomas** [comment]
Lovely!

**Comment by BasThomas** [comment]
Can we remove https://github.com/rnystrom/GitHawk/blob/master/Classes/Views/SegmentedControlCell.swift & friends?

**Comment by rnystrom** [comment]
Lets not just yet in case we need them later

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Pull request merged**